mailpoet/lib/Subscription/Captcha/Validator/BuiltInCaptchaValidator.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace MailPoet\Subscription\Captcha\Validator;
+
+use MailPoet\Models\Subscriber;
+use MailPoet\Subscribers\SubscriberIPsRepository;
+use MailPoet\Subscription\Captcha\CaptchaPhrase;
+use MailPoet\Subscription\Captcha\CaptchaSession;
+use MailPoet\Subscription\SubscriptionUrlFactory;
+use MailPoet\Util\Helpers;
+use MailPoet\WP\Functions as WPFunctions;
+
+class BuiltInCaptchaValidator implements CaptchaValidator {
+
+
+  /** @var SubscriptionUrlFactory  */
+  private $subscriptionUrlFactory;
+
+  /** @var CaptchaPhrase  */
+  private $captchaPhrase;
+
+  /** @var CaptchaSession  */
+  private $captchaSession;
+
+  /** @var WPFunctions  */
+  private $wp;
+
+  /** @var SubscriberIPsRepository  */
+  private $subscriberIPsRepository;
+
+  public function __construct(
+    SubscriptionUrlFactory $subscriptionUrlFactory,
+    CaptchaPhrase $captchaPhrase,
+    CaptchaSession $captchaSession,
+    WPFunctions $wp,
+    SubscriberIPsRepository $subscriberIPsRepository
+  ) {
+    $this->subscriptionUrlFactory = $subscriptionUrlFactory;
+    $this->captchaPhrase = $captchaPhrase;
+    $this->captchaSession = $captchaSession;
+    $this->wp = $wp;
+    $this->subscriberIPsRepository = $subscriberIPsRepository;
+  }
+
+  public function validate(array $data): bool {
+    $isBuiltinCaptchaRequired = $this->isRequired(isset($data['email']) ? $data['email'] : '');
```

Comment:
The default value is `null`, so maybe the else should set to `null` as well?

Reply:
Addressed in [3950064](https://github.com/mailpoet/mailpoet/pull/4196/commits/39500648c313caea6bd3c0ad0064363537b2c060)